src/code/z_message.c

```diff
@@ -64,46 +322,246 @@
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80151BB4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_message/func_80151C9C.s")
+void func_801518B0(GlobalContext* globalCtx, u16 textId, Actor* Actor) {
+    MessageContext* msgCtx;
+    msgCtx = &globalCtx->msgCtx;
+    msgCtx->ocarinaAction = 0xFFFF;
+    func_80150D08(globalCtx, textId);
+    msgCtx->unkActor = Actor;
+    msgCtx->msgMode = 1;
+    msgCtx->unk12023 = 0;
+    msgCtx->unk12024 = 0;
+    globalCtx->msgCtx.ocarinaMode = 0;
+}
+
+void func_80151938(GlobalContext* globalCtx, u16 textId) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    InterfaceContext* interfaceCtx = &globalCtx->interfaceCtx;
+
+    msgCtx->unk11F10 = 0;
+    func_80150D08(globalCtx, (u32)textId);
+    func_80150A84(globalCtx);
+    msgCtx->msgMode = 5;
+    msgCtx->unk12023 = 8;
+    msgCtx->unk12024 = 0;
+
+    if (interfaceCtx->unk_222 == 0) {
+        if (textId != 0x1B93) {
+            func_8011552C(globalCtx, 0x10);
+        } else if (textId != 0xF8) {
+            func_8011552C(globalCtx, 6);
+        }
+    }
+    msgCtx->unk1203C = msgCtx->unk1203A;
+    if (globalCtx->pauseCtx.unk_1F0 != 0) {
+        msgCtx->unk12004 = 0x22;
+        msgCtx->unk12006 = 0x15E;
+        func_80149C18(globalCtx);
+        msgCtx->unk12023 = 1;
+    }
+}
+
+void func_80151A68(GlobalContext* globalCtx, u16 textId) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+
+    msgCtx->unk11F10 = 0;
+    func_80150D08(globalCtx, textId);
+    func_80150A84(globalCtx);
+    func_8015B198(globalCtx);
+    msgCtx->msgMode = 0x45;
+    msgCtx->unk12024 = 0;
+    msgCtx->unk1203C = (msgCtx->unk1203A = (msgCtx->unk1201E = 0));
+    msgCtx->unk12023 = 0x1E;
+
+    // Day/Dawn/Night.. Messages
+    if ((((s32)msgCtx->currentTextId) >= 0x1BB2) && (((s32)msgCtx->currentTextId) < 0x1BB7)) {
+        XREG(74) = 0x6A;
+        XREG(75) = 0;
+        XREG(77) = 0x58;
+        XREG(76) = 0x44;
+    } else {
+        XREG(74) = 0x42;
+        XREG(75) = 0x1E;
+        XREG(77) = 0x3C;
+        XREG(76) = 0x1C;
+        msgCtx->unk11F1A[0] = (msgCtx->unk11F1A[1] = (msgCtx->unk11F1A[2] = 0));
+        Interface_ChangeAlpha(1U);
+    }
+}
+
+void func_80151BB4(GlobalContext* globalCtx, u8 uParm2) {
+    MessageContext* msgCtx = &globalCtx->msgCtx;
+    u8 temp = uParm2;
+
+    if (CHECK_QUEST_ITEM(QUEST_BOMBERS_NOTEBOOK)) {
+        if (((gSaveContext.weekEventReg[D_801C6B28[uParm2] >> 8]) & (u8)D_801C6B28[uParm2]) == 0) {
+            msgCtx->unk120B2[msgCtx->unk120B1] = temp;
+            msgCtx->unk120B1++;
+        }
+    } else if (uParm2 >= 20) {
+        if (((gSaveContext.weekEventReg[D_801C6B28[uParm2] >> 8]) & (u8)D_801C6B28[uParm2]) == 0) {
+            msgCtx->unk120B2[msgCtx->unk120B1] = temp;
+            msgCtx->unk120B1++;
+        }
+    }
+}
+
+u32 func_80151C9C(GlobalContext* globalCtx) {
+    MessageContext* msgCtx;
+    u8 flag;
+
+    msgCtx = &globalCtx->msgCtx;
+    while (1) {
```

Comment:
We usually use `true` / `false` for booleans (but not the `bool` type because it doesn't consistently match)

```suggestion
    MessageContext* msgCtx = &globalCtx->msgCtx;
    u8 flag;

    while (true) {
```

Reply:
done in 41c69051